server/src/main/java/org/opensearch/index/snapshots/blobstore/RemoteStoreShardShallowCopySnapshot.java

```diff
@@ -101,9 +112,21 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
             builder.value(fileName);
         }
         builder.endArray();
+        // We are handling NP check since a cluster can have indexes created earlier which do not have remote store
+        // path type and path hash algorithm in its custom data in index metadata.
+        if (Objects.nonNull(pathType)) {
+            builder.field(PATH_TYPE, pathType.getCode());
+        }
+        if (Objects.nonNull(pathHashAlgorithm)) {
+            builder.field(PATH_HASH_ALGORITHM, pathHashAlgorithm.getCode());
+        }
         return builder;
     }
 
+    /**
+     * This is kept only to avoid breaking changes.
+     */
+    @Deprecated
     public RemoteStoreShardShallowCopySnapshot(
         String snapshot,
```

Comment:
which breaking changes? this will be called from BlobStoreRepository which we must be updating already.

Reply:
You can refer the PR check `Detect Breaking Changes / detect-breaking-change (pull_request) `.

Reply:
Is this exposed to plugins in any way ? Looks like this is an internal class and won't be a breaking change . We should mark it as `InternalApi` .

Reply:
This class has to be marked public api due to below reference in Repository class -
https://github.com/opensearch-project/OpenSearch/blob/9ab31f55e88cc2d023cad6e60a15a0e0e87dac38/server/src/main/java/org/opensearch/repositories/Repository.java#L369-L375
Not changing anything.